src/http2_connection.erl

```diff
@@ -1541,22 +1562,25 @@ good_req_headers(Headers) ->
     end.
 
 no_upper_names(Headers) ->
+    lager:error("Headers: ~p", [Headers]),
     lists:all(
       fun({Name,_}) ->
               NameStr = binary_to_list(Name),
               NameStr =:= string:to_lower(NameStr)
       end,
      Headers).
 
-validate_pseudos([{<<":path">>,_V}|Tail]) ->
-    validate_pseudos(Tail);
-validate_pseudos([{<<":method">>,_V}|Tail]) ->
-    validate_pseudos(Tail);
-validate_pseudos([{<<":scheme">>,_V}|Tail]) ->
-    validate_pseudos(Tail);
-validate_pseudos([{<<":authority">>,_V}|Tail]) ->
-    validate_pseudos(Tail);
-validate_pseudos(DoneWithPseudos) ->
+validate_pseudos(request, [{<<":path">>,_V}|Tail]) ->
```

Comment:
it looks like you don't validate the presence of the required headers, just reject invalid pseudos / wrong order?